libraries/AP_Hygrometer/AP_Hygrometer_UAVCAN.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AP_Hygrometer.h"
+
+#if AP_HYGROMETER_ENABLED
+
+#if HAL_ENABLE_LIBUAVCAN_DRIVERS
+#include <AP_HAL/AP_HAL.h>
+#include "AP_Hygrometer_UAVCAN.h"
+#include <dronecan/sensors/hygrometer/Hygrometer.hpp>
+#include <AP_CANManager/AP_CANManager.h>
+
+#define LOG_TAG "Hygrometer"
+#define C_TO_KELVIN 273.15f
+
+UC_REGISTRY_BINDER(HygrometerCb, dronecan::sensors::hygrometer::Hygrometer);
+
+AP_Hygrometer_UAVCAN::DetectedModules AP_Hygrometer_UAVCAN::_detected_modules[] = {0};
```

Comment:
```suggestion
AP_Hygrometer_UAVCAN::DetectedModules AP_Hygrometer_UAVCAN::_detected_modules[];
```

statics are implicitly zero; save bytes by not initialising